### PR TITLE
feat(websocket): MED-177 replace signet websockets with api call

### PR DESCRIPTION
**Pull request by alicedraillard**
## Describe your changes
signet websocket calls replaced by api calls

## Checklist tests
use Médiacentre main page as usual

## Issue ticket number and link
[MED-177](https://jira.support-ent.fr/browse/MED-177)

## Checklist before requesting a review (magic string, indentation, comment/documentation...)

- [x] I have detailed the tests to do in my feature/fix in order to prevent consequents regressions (must specify in **Checklist tests**)
- [x] I have performed a self-review of my code
- [x] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation (API Doc etc...) - (must specify in **Description** for target version)
- [ ] If it is a consequent feature, I have added thorough tests.
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] Any dependent changes have been added to this project (must specify in **Description**)

### CODE REVIEW

File: src/main/resources/public/ts/controllers/home.ts [hunk 1]
```diff
@@ -78,16 +80,16 @@ class Controller implements IHomeViewModel {
         try {
             await Promise.all([
                 this.syncFavoriteResources(),
-                this.syncTextbooks()
+                this.syncTextbooks(),
```
Comment (alicedraillard) [inline src/main/resources/public/ts/controllers/home.ts:83]: promise.all not used to keep scroll in its original place

File: src/main/resources/public/ts/controllers/main.ts [hunk 2]
```diff
@@ -23,7 +23,7 @@ export interface MainScope extends IRootScopeService {
 	disciplines: Labels;
 	levels: Labels;
 	displayDate (dateToFormat: Date): string;
-	ws: Socket;
+	// ws: Socket;
```
Comment (alicedraillard) [inline src/main/resources/public/ts/controllers/main.ts:26]: will be deleted in later [ticket]([MED-180](https://jira.support-ent.fr/browse/MED-180))

File: src/main/resources/public/ts/controllers/main.ts [hunk 3]
```diff
@@ -90,7 +90,7 @@ class Controller implements MainController {
 	query: any;
 	disciplines: Labels;
 	levels: Labels;
-	ws: Socket;
+	// ws: Socket;
```
Comment (alicedraillard) [inline src/main/resources/public/ts/controllers/main.ts:93]: will be deleted in later [ticket]([MED-180](https://jira.support-ent.fr/browse/MED-180))

File: src/main/resources/public/ts/controllers/main.ts [hunk 4]
```diff
@@ -125,7 +125,7 @@ class Controller implements MainController {
 			}
 		};
 		this.idiom = idiom;
-		this.ws = new Socket();
+		// this.ws = new Socket();
```
Comment (alicedraillard) [inline src/main/resources/public/ts/controllers/main.ts:128]: will be deleted in later [ticket]([MED-180](https://jira.support-ent.fr/browse/MED-180))

File: src/main/resources/public/ts/model/Socket.ts [hunk 5]
```diff
@@ -1,86 +1,86 @@
-import {idiom as lang, toasts} from 'entcore';
```
Comment (alicedraillard) [inline src/main/resources/public/ts/model/Socket.ts:1]: will be deleted in later [ticket]([MED-180](https://jira.support-ent.fr/browse/MED-180))

File: src/main/resources/public/ts/model/resourceBody.model.ts [hunk 6]
```diff
@@ -30,31 +27,28 @@ export class ResourceBody {
...
+    private _structure_name?: string;
+    private _structure_uai?: string;
     private _display_structure_name?: boolean;
     private _user?: string;
     private _id_info: string;
     private _hash: number;
 
     constructor(data: any) {
-        this._id = (data.source == "fr.openent.mediacentre.source.Signet") ? data._id : data.id;
+        this._id = (data.source == "fr.openent.mediacentre.source.Signet") ? undefined : data._id;
```
Comment (alicedraillard) [inline src/main/resources/public/ts/model/resourceBody.model.ts:47]: _id is undefined if resource is a signet

File: src/main/resources/public/ts/model/resourceBody.model.ts [hunk 7]
```diff
@@ -30,31 +27,28 @@ export class ResourceBody {
...
         this._authors = data.authors ? data.authors : [];
         this._date = data.date;
-        this._description = data.description;
+        this._description = data.description ? data.description : null;
```
Comment (alicedraillard) [inline src/main/resources/public/ts/model/resourceBody.model.ts:50]: description is null if it does not exist

File: src/main/resources/public/ts/model/signetBody.model.ts [hunk 8]
```diff
@@ -0,0 +1,216 @@
+import {model} from "entcore";
```
Comment (alicedraillard) [inline src/main/resources/public/ts/model/signetBody.model.ts:1]: model for the body received when getting signets

File: src/main/resources/public/ts/model/resourceBody.model.ts [hunk 9]
```diff
@@ -63,34 +57,31 @@ export class ResourceBody {
...
-        this._action = data.action;
-        this._structure_name = data.structure_name;
-        this._structure_uai = data.structure_uai;
-        this._display_structure_name = data.display_structure_name;
+        this._structure_name = data.structure_name ? data.structure_name : undefined;
+        this._structure_uai = data.structure_uai ? data.structure_uai : undefined;
+        this._display_structure_name = data.display_structure_name ? data.display_structure_name : undefined;
         this._user = data.user;
-        this._id_info = (data.source == "fr.openent.mediacentre.source.Signet") ? data.id : data.id_info;
+        this._id_info = data.id_info ? parseInt(data.id_info) : parseInt(data.id);
```
Comment (alicedraillard) [inline src/main/resources/public/ts/model/resourceBody.model.ts:64]: ensure that id_info is an integer

**Review by qperie — APPROVED** [review]

**Review by flmartineau — APPROVED** [review]

File: src/main/resources/public/ts/model/resourceBody.model.ts (hunk 7 above)
Comment (qpic-cgi) [reply src/main/resources/public/ts/model/resourceBody.model.ts:50]: dans ce cas data.description == null du coup (ou undefined).
pour moi cette gestion n'est pas necessaire.

File: src/main/resources/public/ts/model/resourceBody.model.ts [hunk 10]
```diff
@@ -63,34 +57,31 @@ export class ResourceBody {
         this._plain_text = data.plain_text;
         this._source = data.source;
         this._title = data.title;
-        this._action = data.action;
-        this._structure_name = data.structure_name;
-        this._structure_uai = data.structure_uai;
-        this._display_structure_name = data.display_structure_name;
+        this._structure_name = data.structure_name ? data.structure_name : undefined;
+        this._structure_uai = data.structure_uai ? data.structure_uai : undefined;
+        this._display_structure_name = data.display_structure_name ? data.display_structure_name : undefined;
```
Comment (qpic-cgi) [inline src/main/resources/public/ts/model/resourceBody.model.ts:62]: pas utile ? si ?

File: src/main/resources/public/ts/model/resourceBody.model.ts [hunk 11]
```diff
@@ -63,34 +57,31 @@ export class ResourceBody {
...
             plain_text: this.plain_text,
             source: this.source,
             title: this.title,
-            action: this.action,
-            structure_name: this.structure_name,
-            structure_uai: this.structure_uai,
-            display_structure_name: this.display_structure_name,
+            structure_name: this.structure_name ? this.structure_name : undefined,
+            structure_uai: this.structure_uai ? this.structure_uai : undefined,
+            display_structure_name: this.display_structure_name ? this.display_structure_name : undefined,
```
Comment (qpic-cgi) [inline src/main/resources/public/ts/model/resourceBody.model.ts:84]: pas utile ? si ?

**Review by qpic-cgi — APPROVED** [review]

File: src/main/resources/public/ts/model/resourceBody.model.ts (hunk 11 above)
Comment (alicedraillard) [reply src/main/resources/public/ts/model/resourceBody.model.ts:84]: si parce que sinon ils sont enregistrés comme null en BDD

File: src/main/resources/public/ts/model/resourceBody.model.ts (hunk 7 above)
Comment (alicedraillard) [reply src/main/resources/public/ts/model/resourceBody.model.ts:50]: utilisé pour enregistrer description comme existante mais null en BDD au lieu de ne pas avoir le champ (cf comportement actuel du module)

File: src/main/resources/public/ts/model/resourceBody.model.ts (hunk 10 above)
Comment (alicedraillard) [reply src/main/resources/public/ts/model/resourceBody.model.ts:62]: si parce que sinon ils sont enregistrés comme null en BDD

**Pull request merged**